### PR TITLE
11 get api articles topics

### PR DESCRIPTION
**Pull request by slick-coding**
As per Task 11, this will add topic querying for /api/articles alongside testing and error handling.

### CODE REVIEW

File: __tests__/articles.test.js [hunk 1]
```diff
@@ -122,6 +122,30 @@ describe("GET /api/articles", () => {
...
+                    expect(article.topic).toBe("mitch");
+                });
+            });
+    });
+    test("200: responds with an array of articles with the specified topic, sorted in the correct order", () => {
+        return request(app)
+            .get("/api/articles?topic=mitch&sort_by=votes&order=desc")
+            .expect(200)
+            .then(({ body: { articles } }) => {
+                articles.forEach((article) => {
```
Comment (nc-katherine-hurst) [inline __tests__/articles.test.js:140]: Don't forget to add length checks to avoid false positives in case the array was empty

File: __tests__/articles.test.js [hunk 2]
```diff
@@ -230,6 +254,14 @@ describe("Error Handling", () => {
                     expect(body.msg).toBe("Bad request");
                 });
         });
+        test("404: Responds with an error when given an incorrect topic", () => {
+            return request(app)
+                .get("/api/articles?topic=;DROP TABLES")
+                .expect(404)
+                .then(({ body }) => {
+                    expect(body.msg).toBe("Not found");
+                });
```
Comment (nc-katherine-hurst) [inline __tests__/articles.test.js:263]: Great error handling test 👍

File: __tests__/articles.test.js [hunk 3]
```diff
@@ -122,6 +122,30 @@ describe("GET /api/articles", () => {
...
+            .then(({ body: { articles } }) => {
+                articles.forEach((article) => {
+                    expect(article.topic).toBe("mitch");
+                });
+                expect(articles).toBeSorted({
+                    key: "votes",
+                    descending: true,
+                });
+            });
+    });
```
Comment (nc-katherine-hurst) [inline __tests__/articles.test.js:148]: Could do with one more happy path test to check that if you do a query for a topic which exists but has no articles you get a 200 and an empty array ✅

File: models/articles.models.js [hunk 4]
```diff
@@ -38,7 +42,11 @@ exports.fetchArticles = (sort_by, order) => {
         sqlQuery += "DESC";
     }
 
-    const promises = [db.query(sqlQuery), commentCountLookup()];
+    console.log(sqlQuery)
```
Comment (nc-katherine-hurst) [inline models/articles.models.js:45]: Don't forget to remove console logs from finalised code! 🗑️

**Review by nc-katherine-hurst — COMMENTED** [review]
Great work please find your feedback below 🙂
